### PR TITLE
Added plumed version 2.8.1 including gromacs compatibility

### PR DESCRIPTION
**Pull request by hseara**
Added plumed version 2.8.1

I have also changed the gromacs package that requires proper version compatibility alignment with the new plumed version

### CODE REVIEW

**Review by marvinbernhardt — DISMISSED** [review]

File: var/spack/repos/builtin/packages/gromacs/package.py [hunk 1]
```diff
@@ -170,20 +172,28 @@ class Gromacs(CMakePackage):
...
+    depends_on("plumed@2.8.1+mpi", when="@2020.7+plumed+mpi")
+    depends_on("plumed@2.8.1~mpi", when="@2020.7+plumed~mpi")
+    depends_on("plumed@2.7.5+mpi", when="@2021.5+plumed+mpi")
+    depends_on("plumed@2.7.5~mpi", when="@2021.5+plumed~mpi")
     depends_on("plumed@2.7.3:2.8.0+mpi", when="@2021.4+plumed+mpi")
     depends_on("plumed@2.7.3:2.8.0~mpi", when="@2021.4+plumed~mpi")
     depends_on("plumed@2.7.1:2.7.2+mpi", when="@2021+plumed+mpi")
     depends_on("plumed@2.7.1:2.7.2~mpi", when="@2021+plumed~mpi")
-    depends_on("plumed@2.7.2:2.8+mpi", when="@2020.6+plumed+mpi")
-    depends_on("plumed@2.7.2:2.8~mpi", when="@2020.6+plumed~mpi")
```
Comment (marvinbernhardt) [inline var/spack/repos/builtin/packages/gromacs/package.py]: Wait, does gromacs 2020.6 not work with plumed 2.7.2 anymore? Or why was this changed here?

**Review by tldahlgren — DISMISSED** [review]
Confirmed the two new sha256.

File: var/spack/repos/builtin/packages/gromacs/package.py (hunk 1 above)
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/gromacs/package.py]: That's a great question.

**Comment by hseara** [comment]
Any further problems? or can it be merged? @tldahlgren @marvinbernhardt @junghans

**Review by junghans — APPROVED** [review]
